core/src/main/java/jenkins/security/SecurityListener.java

```diff
@@ -69,13 +69,27 @@ protected void failedToAuthenticate(@Nonnull String username){}
      */
     protected void loggedIn(@Nonnull String username){}
 
+    /**
+     * Fired when a new user account has been created.
+     *
+     * @param username the user
+     */
+    protected void userCreated(@Nonnull String username) {}
+
     /**
      * Fired when a user has failed to log in.
      * Would be called after {@link #failedToAuthenticate}.
      * @param username the user
      */
     protected void failedToLogIn(@Nonnull String username){}
 
+    /**
+     * Fired when new user account creation fails.
+     *
+     * @param username the user
+     */
+    protected void failedToCreateUser(@Nonnull String username){}
```

Comment:
Now I know this isn't the exact pattern that exists here, but what would you think about adding a second parameter like `@CheckForNull Exception cause` so that the exception that would otherwise be thrown later can be handled here as well. That wouldn't be required for our specific use case, though I'm curious if anyone thinks that would be useful to include anyways.

Reply:
By the way, if we decide to add listeners that allow for an exception as well, we can likely add those later.